crates/bevy_render/src/render_resource/bind_group.rs

```diff
@@ -291,6 +291,49 @@ impl Deref for BindGroup {
 ///   binding_array<StandardMaterialUniform>` and accessible as
 ///   `material_array[material_indices[slot].material]`.
 ///
+/// ## `data(BINDING_INDEX, ConvertedShaderType, binding_array(BINDING_INDEX))`
+///
+/// * This is very similar to `uniform(BINDING_INDEX, ConvertedShaderType,
+///   binding_array(BINDING_INDEX)` and in fact is identical if bindless mode
+///   isn't being used. The difference is that, in bindless mode, the `data`
+///   attribute produces a single buffer containing an array, not an array of
+///   buffers. For example, suppose you had the following declaration:
+///
+/// ```ignore
+/// #[uniform(0, StandardMaterialUniform, binding_array(10))]
+/// struct StandardMaterial { ... }
+/// ```
+///
+/// This will produce a binding matching the following WGSL declaration:
+///
+/// ```wgsl
+/// @group(2) @binding(10) var<storage> material_array: binding_array<StandardMaterial>;
+/// ```
+///
+/// On the other hand, if you write this declaration:
+///
+/// ```ignore
+/// #[data(0, StandardMaterialUniform, binding_array(10))]
+/// struct StandardMaterial { ... }
+/// ```
+///
+/// Then Bevy produces a binding that matches this WGSL declaration instead:
+///
+/// ```wgsl
+/// @group(2) @binding(10) var<storage> material_array: array<StandardMaterial>;
```

Comment:
In the bindless case only, right? We perhaps need to illustrate both here so people can copy and paste?

Reply:
Good point. I clarified that this example is only for bindless mode and added a new example illustrating how it looks in non-bindless mode.